packages/xo-server/src/api/pool.mjs

```diff
@@ -43,6 +43,10 @@ export async function set({
     crashDumpSr !== undefined &&
       pool.$call('set_crash_dump_SR', crashDumpSr === null ? Ref.EMPTY : crashDumpSr._xapiRef),
   ])
+  if (nameLabel !== undefined || nameDescription !== undefined) {
+    const serverId = this.getXenServerIdByObject(pool.uuid, 'pool')
+    await this.updateXenServer(serverId, { poolNameDescription: nameDescription, poolNameLabel: nameLabel })
+  }
```

Comment:
A pool's name may be changed by another client. Also, I think the most common use case for this feature would be to get the name of the pool as soon as the user connects to the server. Updating it when the pool's name_label changes is almost a bonus because it doesn't change that often.

So I think a better approach for this would probably be a sort of hook on the XAPI events to catch any changes of the pools' names and update their servers accordingly.

Reply:
What I did does display the name of the pool when a server is added or connected to.

But you're right, I did not realize the pool's name could be changed by another client.